src/useCapabilitiesSafe.ts

```diff
@@ -0,0 +1,36 @@
+import type { Chain } from 'viem';
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+import type { WalletCapabilities } from './types';
+
+export function useCapabilitiesSafe({
+  chain,
+}: {
+  chain: Chain;
+}): WalletCapabilities {
+  const { connector, isConnected } = useAccount();
+
+  // Metamask doesn't support wallet_getCapabilities
+  const isMetamaskWallet = connector?.id === 'io.metamask';
+  const enabled = isConnected && !isMetamaskWallet;
```

Comment:
optionally we can also just check for `error` returned from `useCapabilities` but there will still be an error displayed in the console for unsupported RPCs - IMO this is a more scalable solution than manually excluding individual connector IDs

Reply:
cc @alissacrane-cb on Tanstack query best practices